.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,308 @@
+
+### 3. Required Reading
+
+- [Original Satoshi Whitepaper](https://bitcoin.org/bitcoin.pdf) - This is the
+  white paper that started it all.  Having a solid foundation to build on will
+  make the code much more comprehensible.
+- [Don't block the Event Loop](https://nodejs.org/en/docs/guides/dont-block-the-event-loop/)
+
+<a name="DevelopmentPractices" />
+
+### 4. Development Practices
+
+Developers are expected to work in their own trees and submit pull requests when
+they feel their feature or bug fix is ready for integration into the  master
+branch.
+
+<a name="ShareEarly" />
+
+### 4.1 Share Early, Share Often
+
+We firmly believe in the share early, share often approach.  The basic premise
+of the approach is to announce your plans **before** you start work, and once
+you have started working, craft your changes into a stream of small and easily
+reviewable commits.
+
+This approach has several benefits:
+
+- Announcing your plans to work on a feature **before** you begin work avoids
+  duplicate work
+- It permits discussions which can help you achieve your goals in a way that is
+  consistent with the existing architecture
+- It minimizes the chances of you spending time and energy on a change that
+  might not fit with the consensus of the community or existing architecture and
+  potentially be rejected as a result
+- Incremental development helps ensure you are on the right track with regards
+  to the rest of the community
+- The quicker your changes are merged to master, the less time you will need to
+  spend rebasing and otherwise trying to keep up with the main code base
+
+
+<a name="CommonMistakes" />
+
+### 4.2 CommonMistakes
+
+Some common mistakes that you might stumble across:
+
+- Not using sync/async correctly, see: [Required Reading #2](#ReqReading)
+
+<a name="Testing" />
+
+### 4.3 Testing
+
+One of the major design goals of all core bcoin packages is to aim for complete
+test coverage.  This is financial software so bugs and regressions can cost
+people real money.  For this reason every effort must be taken to ensure the
+code is as accurate and bug-free as possible.  Thorough testing is a good way to
+help achieve that goal.
+
+Unless a new feature you submit is completely trivial, it will probably be
+rejected unless it is also accompanied by adequate test coverage for both
+positive and negative conditions.  That is to say, the tests must ensure your
+code works correctly when it is fed correct data as well as incorrect data
+(error paths).
+
+A quick summary of test practices follows:
+- All new code should be accompanied by tests that ensure the code behaves
+  correctly when given expected values, and, perhaps even more importantly, that
+  it handles errors gracefully
+- When you fix a bug, it should be accompanied by tests which exercise the bug
+  to both prove it has been resolved and to prevent future regressions
+
+<a name="CodeDocumentation" />
+
+### 4.4 Code Documentation and Commenting
+
```

Comment:
Maybe we should indicate that we expect functions and methods to be documented using the jsdoc standards for doc generation

Reply:
with a link for reference

Reply:
Cool, we could actually have a `require-jsdoc` to `eslintrc`, to enforce this.

Reply:
`require-jsdoc` seems to be too restrictive, it shows ~250 failures on the current codebase. Not adding it for now.